app/react/components/TLSFieldset/TLSFieldset.tsx

```diff
@@ -76,21 +82,37 @@ export function TLSFieldset() {
       )}
     </>
   );
+
+  function handleChange(partialValue: Partial<TLSConfig>) {
+    onChange(partialValue);
+  }
 }
 
 const MAX_FILE_SIZE = 5_242_880; // 5MB
 
-function certValidation() {
+type CertValidationContext = {
+  optionalCertValidation: boolean;
+};
+
+function certValidation(certValidation?: CertValidationContext) {
```

Comment:
can I suggest to simplify it? (certValidation recurres too much :) )
```suggestion
function certValidation(optional?: boolean) {

...
function tlsConfigValidation({optionalCerts}: {optionalCerts?: boolean} = {})

```